llvm/lib/Target/AMDGPU/AMDGPUPromoteAlloca.cpp

```diff
@@ -1563,18 +1563,16 @@ bool AMDGPUPromoteAllocaImpl::tryPromoteAllocaToLDS(AllocaInst &I,
         Value *Size = Intr->getArgOperand(0);
         ArgTy.emplace_back(Offset->getType());
         Args.emplace_back(Size);
-        Args.emplace_back(Offset);
       } else if (Intr->getIntrinsicID() == Intrinsic::invariant_end) {
         Value *InvariantPtr = Intr->getArgOperand(0);
         Value *Size = Intr->getArgOperand(1);
         ArgTy.emplace_back(Offset->getType());
         Args.emplace_back(InvariantPtr);
```

Comment:
I eventually didn't make it an initialize list because that doesn't look very "fit". I figure one extra line is fine.

Reply:
You can still initialize ArgTys with Offset->getType(), it's the first entry in every case